### ISSUE TITLE
Handling of config structs with tagged struct fields

### ISSUE THREAD

**Issue by hariso** [ConduitIO/ecdysis#2]
In #1 we added support for parsing nested structs. We found that the following case is possible and needs to be properly handled:

```go
type Config struct {
  DBConfig DBConfig `long:"database config"` 
}
type DBConfig struct {
  URL string `long:"db url"`
}
```

**Comment by lovromazgon** [comment]
IIUC this issue is about adding a prefix to nested flags, correct? For example, these structs:

```
type Config struct {
  DBConfig DBConfig `long:"database"` 
}
type DBConfig struct {
  URL string `long:"url"`
}
```

... would result in the flag `-database.url=foo`, right?

**Comment by hariso** [comment]
@lovromazgon Yes, that's one of the options. Another is to just disallow that usage at all. My intention with this issue is to clarify which do we want to use and be consistent with that.
